### PR TITLE
Remove protective styling on subreddit style checkbox

### PR DESCRIPTION
**Pull request by andytuba**
The mods can have nice things again. I'll brace for educating the userbase "yes, it might disappear in some subreddits, here are some other methods."

### CODE REVIEW

File: lib/modules/showImages.js [hunk 1]
```diff
@@ -1670,24 +1667,6 @@ addModule('showImages', {
...
-		RESUtils.debounce('handleSRStyleToggleVisibility', 50, function() {
-			var toggleEle = modules['styleTweaks'].styleToggleContainer;
-			if (!toggleEle) return;
-			var imageElems = image ? [image] : document.querySelectorAll('.RESImage');
-
-			for (var i = 0; i < imageElems.length; i++) {
-				var imageEle = imageElems[i];
-				var imageID = imageEle.getAttribute('id');
-
-				if (RESUtils.doElementsCollide(toggleEle, imageEle, 15)) {
```
Comment (erikdesjardins) [inline lib/modules/showImages.js]: `RESUtils.doElementsCollide` is a candidate for removal (I've never seen it anywhere else)

File: lib/modules/styleTweaks.js [hunk 2]
```diff
@@ -486,30 +485,9 @@ addModule('styleTweaks', {
 		this.styleToggleLabel.textContent = 'Use subreddit style ';
 		this.styleToggleContainer.appendChild(this.styleToggleLabel);
 
-		this.protectElement(this.styleToggleContainer, 'display: block !important;');
-		this.protectElement(this.styleToggleCheckbox, 'display: inline-block !important;');
-		this.protectElement(this.styleToggleLabel, 'display: inline-block !important; margin-left: 4px !important;');
-
-		this.setSRStyleToggleVisibility(true); // no source
-	},
-	srstyleHideLock: RESUtils.createMultiLock(),
```
Comment (erikdesjardins) [inline lib/modules/styleTweaks.js]: Ditto for `createMultiLock`

File: lib/modules/styleTweaks.js [hunk 3]
```diff
@@ -535,114 +508,6 @@ addModule('styleTweaks', {
...
-
-		// great, now people are still finding ways to hide this.. these extra declarations are to try and fight that.
-		// Sorry, subreddit moderators, but users can disable all subreddit stylesheets if they want - this is a convenience
-		// for them and I think taking this functionality away from them is unacceptable.
-
-		var zIndex = 'z-index: ' + (visible ? ' 2147483646' : 'auto') + ' !important;';
-
-		self.protectedElements.each(function(index, element) {
-			var elementCss = element.getAttribute('data-res-css') || '';
-			var css = RESUtils.baseStyleProtection + zIndex + elementCss;
```
Comment (erikdesjardins) [inline lib/modules/styleTweaks.js]: Ditto for `baseStyleProtection` (and there are some helpers used to generate it that could also be removed)

**Comment by erikdesjardins** [comment]
LGTM with those helpers removed

File: lib/modules/showImages.js (hunk 1 above)
Comment (andytuba) [inline lib/modules/showImages.js]: :ballot_box_with_check:

File: lib/modules/styleTweaks.js (hunk 2 above)
Comment (andytuba) [inline lib/modules/styleTweaks.js]: :ballot_box_with_check:

File: lib/modules/styleTweaks.js (hunk 3 above)
Comment (andytuba) [inline lib/modules/styleTweaks.js]: :ballot_box_with_check:

**Pull request merged**